Sparkbox Sprint 62

lunarfusion opened the pull request:
# READY FOR REVIEW

## Summary
* #1089
* #1087 
* #1097 
* #1095 
* #1098 
* #1093

TASK LIST:
- [x]  Add compiled CSS & JS
- [x]  Run BackstopJS visual regression tests
- [x]  Ensure Code Climate passes against the develop branch. https://codeclimate.com/github/SU-HSDO/suhumsci/pull/1086
- [x]  Assign to Mike for review

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. `npm run test` should pass in the theme directory
2. If you are interested in the testing steps for associated PRs, please take a look at the individual PR's description and "Steps to Test" section.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat: use we_megamenu process hook to add private page icon before private pages, style lock icon in megamenu

Commit:
add newline to eof

Commit:
fix: resize lock icon and fix padding

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

lunarfusion committed:
STN-974: Move Mega Menu SCSS (#1087)

* feat(STN-974): moved megamenu css into theme

Blocked megamenu CSS and bootstrap from loading from the module.  Moved compiled megamenu CSS from we_megamenu_backend.css into _vendor.we-megamenu.css. Removed Quicksand and Fontawesome styles. Removed unused styles related to animations and color variations. Linted CSS.

* feat(STN-974): moved mega menu base css into css directory

* feat(STN-974) move megamenu base CSS into themes

* feat(STN-974): add mega menu base CSS library via theme hook

* feat(STN-974): removed unwanted CSS from megamenu base CSS

also updated comment on colorful theme file, reinforced no underlines on level 1 and level 3 links

* feat(STN-974): Updated yml and theme files to correctly reference wemegamenu

* feat(STN-974): moved megamenu library after JS

Megamenu library loading before JS libraries was causing megamenu not to respond correctly to JS and resulting in interference of megamenu CSS in standard menu function. Result was a hidden standard menu on mobile. This corrects the issue.

* chore: compiled CSS

Co-authored-by: Kasey Bonifacio <kasey@heysparkbox.com>

lunarfusion committed:
fix(STN-994): scss for private page icon (#1097)

* fix(STN-994): scss for private page icon

* chore: add complied css

Co-authored-by: Melissa Piper <mpiper@heysparkbox.com>

Commit:
fix: keep expand/collapse button in place when clicked (#1095)

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

kaseybon committed:
Chore: Sparkbox Cleanup (#1098)

* chore: update the humsci basic pull request checklist

* docs: remove old behat testing documentation

* docs: update backstop visual regression documentation

* chore: update backstop dependency

* chore: udpate browserslist data and add documentation

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

Commit:
STN-982: Mega Menu On-Click Modification (#1093)

* feat: force we_megamenu to use click-action instead of hover-action

* remove duplicate scss

* chore: fix we-megamenu scss conflicts from rebase

* chore: add compiled CSS, re-copy base megamenu css to resolve style issues

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

kaseybon committed:
chore: compiled CSS

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

lunarfusion committed:
fix(sparkbox-sprint-62): corrected linting errors

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

pookmish commented on docroot/themes/humsci/humsci_colorful/humsci_colorful.theme:
```diff
@@ -23,6 +23,13 @@ function humsci_colorful_preprocess_html(&$vars) {
+  $use_megamenu = theme_get_setting('megamenu_toggle');
+  if ($use_megamenu) {
+    // Load base megamenu CSS when megamenu is enabled
+    $vars['#attached']['library'][] =  'humsci_colorful/megamenu';
```
This should be added at the block layer using a `preprocess_block__we_megamenu` I believe. Then you don't need to have the condition if the theme has it turned on

pookmish replied:
alternatively, you could alter the library using a `hook_library_info_alter` and add the dependency

pookmish commented on docroot/themes/humsci/humsci_traditional/humsci_traditional.theme:
```diff
@@ -30,6 +30,13 @@ function humsci_traditional_preprocess_html(&$vars) {
+  $use_megamenu = theme_get_setting('megamenu_toggle');
+  if ($use_megamenu) {
+    // Load base megamenu CSS when megamenu is enabled
+    $vars['#attached']['library'][] =  'humsci_traditional/megamenu';
```
This should be added at the block layer using a `preprocess_block__we_megamenu` I believe. Then you don't need to have the condition if the theme has it turned on

lunarfusion replied:
Please take a look at the latest commit. I've changed it to we_megamenu_preprocess_block and tested Traditional and Colorful with the mega menu disabled and enabled. Appears to load the base CSS library only when it's enabled as hoped.

pookmish replied:
Now this function will never run. A hook does not begin like this. Target the specific megemu block and rove the theme get settings line

lunarfusion committed:
fix(sparkbox-sprint-62): change mega menu library from preprocess html to preprocess block

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

Commit:
fix: update block preprocess hook for megamenu in traditional and colorful themes

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

pookmish reviewed (approved):
Thank you. This is the correct way.
